CMakeLists.txt

```diff
@@ -134,7 +134,7 @@ ENDIF()
 
 IF(WITH_VIGRANUMPY)
     set(WITH_BOOST "ON")
-    # set(WITH_BOOST_COMPONENTS ${WITH_BOOST_COMPONENTS} "python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+    set(WITH_BOOST_COMPONENTS ${WITH_BOOST_COMPONENTS} "python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
```

Comment:
`FIND_LIBRARY` is configured such that it will prefer the `python-py*` variant if it exists, but falls back to alternative conventions otherwise. It can indeed happen that it picks an incompatible boost_python version this way, but the WITH_BOOST_COMPONENTS approach just fails the other way around - it may miss a perfectly valid library.